api/src/main/java/org/apache/cloudstack/api/command/user/template/CreateTemplateCmd.java

```diff
@@ -251,26 +267,33 @@ public long getEntityOwnerId() {
             }
         }
 
-        if(projectId != null){
+        Account accountToUse = callingAccount;
+        if (accountName != null && domainId != null) {
+            try {
+                accountToUse = _accountService.getAccount(_accountService.finalyzeAccountId(accountName, domainId, projectId, true));
+
+            } catch (Exception exception) {
+                s_logger.info("Unable to find accountId associated with accountName=" + accountName + " and domainId="
+                      + domainId + " using account=" + accountToUse);
+            }
+        } else if (projectId != null) {
             final Project project = _projectService.getProject(projectId);
             if (project != null) {
```

Comment:
DO we need this block as `finalyzeAccountId` already seems to be taking care of the projectid

Reply:
Right, it's not needed anymore. I have removed this block of code.